river/tree/_attribute_observer/__init__.py

```diff
@@ -20,4 +26,6 @@
     "NumericAttributeClassObserverGaussian",
     "NumericAttributeClassObserverHistogram",
     "NumericAttributeRegressionObserver",
+    "NumericAttributeRegressionQuantizerObserver",
```

Comment:
Our nemesis, long class names.

What about `NumericAttributeRegQuantizerObserver`. 

However, we might need something even shorter since we will expose the obersvers to the user @_@

Reply:
I like the terminology "attribute observer" because, well, it sounds pretty cool :) 

So, I prefer it to splitter. But I don't have strong feelings. We might go with either:

- splitter 
- AO (for Attribute Observer, as I did in my paper)

We also need to define/revise our naming pattern in this upcoming PR. Maybe something like:

- QOSplitter
- EBSTSplitter

or

- AORegQO
- AORegEBST 

Any thoughts, @jacobmontiel and @MaxHalford ?

Reply:
I don't like AO because it's not an obvious acronym. I like splitter :)

Reply:
I see your point. I'm slowly starting to like such a naming scheme:

> - QOSplitter
> - EBSTSplitter

Time heals everything, hahaha

Reply:
My two cents. Both terms are valid.

- **Attribute Observers** is the name traditionally used in stream learning literature. 
- **Splitters** is the counterpart used in batch learning (found in `scikit-learn` trees).

What about a compromise:

- We can use "Attribute Observers" in the documentation: "Attribute Observers (splitters) are..."
- We use "splitters" in the implementation as they are shorter, e.g. `QOSplitter`.
- We can have a module `attribute_observers` that contains `XXXXSplitter` classes.
- This way we enforce the fact that they represent the same thing.

Reply:
I am ok with only keeping Attribute Observer in the documentation, to make them familiar to the stream gardeners, like me. Maybe, calling the module `splitter` will make it familiar to those used to `scikit-learn` and also shorten the name.